runnable/extra-files/dexcept.h

```diff
@@ -0,0 +1,63 @@
+// D exceptions to by caught by C++
+
+#pragma once
+
+namespace D {
+
+struct string
+{
+    size_t length;
+    const char* ptr;
+};
+
+class Object;
+
+extern "C" {
+    string _d_toString(Object* o);
+}
+
+class Object
+{
+    void* __monitor;
+
+    virtual void _classinfo_data() = 0;
+
+protected:
+    // don't call directly, ABI mismatch

```

Comment:
> extern(D)!

Yeah, stupid me, completely forgot about that for a moment. The fever I had the last few days must have damaged my brain...

Do you think this alternative C++ mangling + `_d_toString` should be merged into druntime as-is? Or is it meant as prototype for a standardized upstream approach?